util/src/main/java/tc/oc/pgm/util/collection/ImmutableMaterialSet.java

```diff
@@ -0,0 +1,81 @@
+package tc.oc.pgm.util.collection;
+
+import com.google.common.collect.ForwardingSet;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.EnumSet;
+import java.util.Set;
+import org.bukkit.Material;
+
+/**
+ * Highly efficient immutable container for {@link Material}s. Useful for sharing large material
+ * lists.
+ */
+public class ImmutableMaterialSet extends ForwardingSet<Material> {
```

Comment:
Do we actually need this? Can we just use a `Set<Material>` wherever this is used?